notebook/notebookapp.py

```diff
@@ -1421,7 +1421,10 @@ def init_webapp(self):
             # SSL may be missing, so only import it if it's to be used
             import ssl
             # Disable SSLv3 by default, since its use is discouraged.
-            ssl_options.setdefault('ssl_version', ssl.PROTOCOL_TLSv1)
+            ssl_options.setdefault(
+                'ssl_version', 
+                ssl.PROTOCOL_TLS if hasattr(ssl, PROTOCOL_TLS) else ssl.PROTOCOL_SSLv23
```

Comment:
`hasattr()` takes a string as the second argument. Though it might be neater to use `getattr()` with a default.

Reply:
> hasattr() takes a string as the second argument. 
you're right. thanks for catch that up. I fixed that it is done for review. 

> Though it might be neater to use getattr() with a default.
sorry I didn't understand how to use this default

Reply:
Like `getattr(ssl, 'PROTOCOL_TLS', ssl.PROTOCOL_SSLv23)`. But the hasattr version is fine too.

Reply:
oh I see ... that is much better thanks. I forgot getattr has default argument. thanks. I will change that.